core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java

```diff
@@ -1512,12 +1723,16 @@ private void processBind(long lo, long msgLimit, @Transient SqlCompiler compiler
                         // good to go
                         for (int i = 0; i < columnCount; i++) {
                             lo += Short.BYTES;
-                            activeSelectColumnTypes.setQuick(i, toColumnBinaryType(getShortUnsafe(lo), m.getColumnType(i)));
+                            final short code = getShortUnsafe(lo);
+                            activeSelectColumnTypes.setQuick(2 * i, toColumnBinaryType(code, m.getColumnType(i)));
+                            activeSelectColumnTypes.setQuick(2 * i + 1, 0);
                         }
                     } else if (columnFormatCodeCount == 1) {
+                        lo += Short.BYTES;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of `Integer` multiplication is cast into a larger type. If you think the result of the multiplication could exceed the size of an `Integer`, consider casting the arguments of the multiplication to avoid arithmetic overflows. For more information, see [CWE-190](https://cwe.mitre.org/data/definitions/190.html).